src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,199 @@
+    } catch (IOException ex) {
+      throw new IOException("Could not create/file with given fileName");
+    } catch (XMLStreamException ex) {
+
+    }
+    return fromXML(x, xmlr);
+  }
+
+  public static void copyFromXML(FObject obj, XMLStreamReader reader) throws XMLStreamException {
+    try {
+      PropertyInfo prop = null;
+      while ( reader.hasNext() ) {
+        int eventType;
+        eventType = reader.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_ELEMENT:
+            ClassInfo cInfo = obj.getClassInfo();
+            prop = (PropertyInfo) cInfo.getAxiomByName(reader.getLocalName());
+            if ( prop == null ) {
+              throw new XMLStreamException("Could not find property " + reader.getLocalName());
+            }
+            break;
+          case XMLStreamConstants.CHARACTERS:
+            if ( prop != null ) {
+              prop.setFromString(obj, reader.getText());
+              prop = null;
+            }
+            break;
+          case XMLStreamConstants.END_ELEMENT:
+            if ( reader.getLocalName().equals("object") ) {
+              return;
+            }
+            break;
+        }
+      }
+    } catch (XMLStreamException ex) {
+      throw new XMLStreamException("Premature end of xml file");
+    }
+  }
+
+  public static void toXML(List<FObject> objList, XMLStreamWriter xmlStreamWriter) {
+    try {
+      xmlStreamWriter = new IndentingXMLStreamWriter(xmlStreamWriter);
+      Iterator i = objList.iterator();
+      xmlStreamWriter.writeStartDocument();
+      //Root tag
+      xmlStreamWriter.writeStartElement("objects");
+
+      while ( i.hasNext() ) {
+        toXML((FObject) i.next(), xmlStreamWriter);
+      }
+
+      xmlStreamWriter.writeEndElement();
+      xmlStreamWriter.writeEndDocument();
+
+    } catch (XMLStreamException ex) {
+    }
+  }
+
+  public static void toXML(FObject obj, XMLStreamWriter xmlStreamWriter) throws XMLStreamException {
+    try {
+      xmlStreamWriter.writeStartElement("object");
+      xmlStreamWriter.writeAttribute("class", obj.getClass().toString().replaceAll("class ", ""));
+      writeObjectProperties(obj, xmlStreamWriter);
+      xmlStreamWriter.writeEndElement();
+    } catch (XMLStreamException ex) {
+      throw new XMLStreamException("Error while writing xml file");
+    }
+  }
+
+  // Write properties from given FObject
+  public static void writeObjectProperties(FObject obj, XMLStreamWriter writer) {
+    ClassInfo cInfo = obj.getClassInfo();
+    List props = cInfo.getAxioms();
+    Iterator propItr = props.iterator();
+
+    try {
+      while ( propItr.hasNext() ) {
+        PropertyInfo prop = (PropertyInfo) propItr.next();
+        Object value = prop.f(obj);
+        if (prop.getTransient() ) continue;
+        if ( value != null && value != "" ) {
+          writer.writeStartElement(prop.getName());
+          // Case for date
+          if ( value instanceof java.util.Date ) {
```

Comment:
There shouldn't be any special casing for Date or any other type. If you need to do something special, rely on polymorphism from the PropertyInfo.